bundles/org.eclipse.osgi/container/src/org/eclipse/osgi/internal/location/BasicLocation.java

```diff
@@ -152,6 +164,11 @@ public synchronized boolean set(URL value, boolean lock, String lockFilePath) th
 		location = value;
 		if (property != null)
 			configValues.setConfiguration(property, location.toExternalForm());
+		if (serviceRegistration != null) {
+			Dictionary<String, Object> properties = serviceRegistration.getReference().getProperties();
+			properties.put("url", location.toExternalForm()); //$NON-NLS-1$
+			serviceRegistration.setProperties(properties);
```

Comment:
Should surround the call to `setProperties` with a `catch (IllegalStateException e)`, I would just ignore the exception and continue on in this case.

Reply:
Done